features/feature_case/kernel_library/kernel_library.cpp

```diff
@@ -0,0 +1,168 @@
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include <string>
+#include <cuda_runtime_api.h>
+#include <cuda.h>
+#include "init.hpp"
+
+#define VEC_LENGTH 128
+#define SEED      59
+
+static int load_type;
+static int invoke_type;
+
+static void loadModule(CUmodule &module, const char *ptxfile) {
+if (load_type==0) {
+  checkErrors( cuModuleLoad(&module,ptxfile) );
+} else if (load_type==1 || load_type==2) {
+  std::ifstream      ifile(ptxfile, std::ios::in | std::ios::binary);
+  std::stringstream  strStream; 
+  std::string        content;
+
+  // put file into a buffer for loading
+  strStream << ifile.rdbuf();
+  content = strStream.str();
+  
+  if (load_type==1) {
+    checkErrors( cuModuleLoadData(&module,content.c_str()) );
+  } else {
+    /*
+    const unsigned int jitNumOptions = 1;
+    CUjit_option  jitOptions[jitNumOptions];
+    void         *jitOptVals[jitNumOptions];
+  
+    // set up wall clock time
+    jitOptions[0] = CU_JIT_WALL_TIME;
+    */
+
+    checkErrors( cuModuleLoadDataEx(&module,content.c_str(),
+                                    // jitNumOptions, jitOptions, (void **)jitOptVals)
+                                    0, NULL, NULL)
+                );
+
+    /*
+    unsigned long long opt_int = (unsigned long long) jitOptVals[0];
+    float jitTime = *((float *) &opt_int);
+      
+    std::cout << "\tcompile-time: " << jitTime << "ms\n";
+    */
+  }
+}
+ 
+}
+
+int run_tests() {
+  static int cnt;
+
+  for (load_type=0; load_type<3; load_type++) {
+    for (invoke_type=0; invoke_type<2; invoke_type++) {
+      int seed = SEED;
+
+      // initialize
+      init_CUDA();
```

Comment:
`init_CUDA` is called every iteration, is this this intentional?

Reply:
init_CUDA is unnecessary.  Call removed.